apm: add 8.12.0 known issue about fleet ingest pipeline bug (#3597)
elastic/observability-docs: docs/en/observability/apm/known-issues.asciidoc

// Link to fix?
A fix was released in 8.11.2: https://github.com/elastic/kibana/pull/171712[elastic/kibana#171712].

*traces-apm@custom ingest pipeline applied to certain data streams unintentionally* +
_APM Server versions: 8.12.0_ +

// Describe the conditions in which this issue occurs
If you're using the Elastic APM Server v8.12.0,
// Describe the behavior of the issue

> a bug has been introduced, so that `traces-apm@custom` ingest pipeline is now additionally applied to data streams `traces-apm.sampled-*`
> and `traces-apm.rum-*`, and applied twice for `traces-apm-*`. This impacts users with a non-empty `traces-apm@custom` ingest pipeline.

Reviewer: ```suggestion
the `traces-apm@custom` ingest pipeline is now additionally applied to data streams `traces-apm.sampled-*`
and `traces-apm.rum-*`, and applied twice for `traces-apm-*`. This bug impacts users with a non-empty `traces-apm@custom` ingest pipeline.
```

Author: thanks, I always appreciate these suggestions!

> If you rely on this unintended behavior in 8.12.0, please rename your pipeline to `traces-apm.package@custom` to preserve this behavior in 8.12.1.

Author: ```suggestion
If you rely on this unintended behavior in 8.12.0, please rename your pipeline to `traces-apm.package@custom` to preserve this behavior in later versions.
```
^ will this be clearer that it applies to 8.12.1+? or is it already implied?

Reviewer 2: The fix actually ended up using `.integration` and not `package` as suffix, see https://github.com/elastic/package-spec/pull/704 and https://github.com/elastic/kibana/pull/175448/files.

Author: good catch! I copied from PR description without noticing it

> If you rely on this unintended behavior in 8.12.0, please rename your pipeline to `traces-apm.package@custom` to preserve this behavior in later versions.

Author: ```suggestion
If you rely on this unintended behavior in 8.12.0, please rename your pipeline to `traces-apm.integration@custom` to preserve this behavior in later versions.
```
